scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_impl.cpp.tpl

```diff
@@ -20,46 +20,41 @@ The notice below applies to the generated files.
 
 /** if upcastable **/
 
-using Key = /*= struct | user_fullname =*/::Key;
-
-Key KeyedObjectTraits<UpcastablePointer</*= struct | user_fullname =*/>>::key(
-  const UpcastablePointer</*= struct | user_fullname =*/> &Obj)
-{
+using /*= struct.name =*/Key = /*= struct | user_fullname =*/::Key;
+using U/*= struct.name =*/ = /*= struct.namespace =*/::Upcastable/*= struct.name =*/;
```

Comment:
What is this used for? Just `U` as a prefix seems a bit confusing.

Reply:
1. It was broken, since these typedefs didn't depend on the struct name, it could only work if there is at most one polymorphic struct per TTG invocation.

2. `U` stands for `Upcastable`, I wanted to fit at least `Up`, but the space is pretty limited.

Reply:
Is the `U`-`using` for internal use only or is it supposed to be available to users too?

Reply:
This is the `Impl.cpp`'s template, so internal only.